src/main/java/ch/jalu/configme/properties/EnumSetProperty.java

```diff
@@ -0,0 +1,31 @@
+package ch.jalu.configme.properties;
+
+import ch.jalu.configme.properties.types.EnumSetPropertyType;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * Introduce an EnumSetProperty that goes along with EnumSetPropertyType.
+ *
+ * @param <E> enum
+ */
+public class EnumSetProperty<E extends Enum<E>> extends SetProperty<Set<E>> {
+
+    public EnumSetProperty(String path, Class<E> enumClass, EnumSet<E> defaultValue) {
+        super(path, new EnumSetPropertyType(enumClass), defaultValue);
+    }
+
+    public EnumSetProperty(String path, Class<E> enumClass, E... defaultValue) {
+        super(path, new EnumSetPropertyType(enumClass), castToEnumSet(enumClass, defaultValue));
```

Comment:
SetProperty has a few constructors. This one being called here uses a property type that defines how a single entry in the set is being converted. EnumSetPropertyType is the type for the _whole_ set though—so we need to use the constructor that takes a property type as first argument. It's confusing, but Java kind of forces us to distinguish them by order (because there's no other way)... 

```java
    public SetProperty(@NotNull String path, @NotNull PropertyType<E> entryType, ...)
```
vs.
```java
    protected SetProperty(@NotNull PropertyType<Set<E>> type, @NotNull String path, ...)
```

What you want to do here is this:
```suggestion
        super(new EnumSetPropertyType(enumClass), path, castToEnumSet(enumClass, defaultValue));
```